canal/canal.go

```diff
@@ -481,17 +481,25 @@ func (c *Canal) prepareSyncer() error {
 	if strings.Contains(c.cfg.Addr, "/") {
 		cfg.Host = c.cfg.Addr
 	} else {
-		seps := strings.Split(c.cfg.Addr, ":")
```

Comment:
I think we can use https://pkg.go.dev/net#SplitHostPort

Reply:
Good idea, let me check this